### PR TITLE
Internal: Extract Web-CLI to external module [ED-6114]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: modules/web-cli/assets/js/api.js [hunk 1]
```diff
@@ -27,8 +27,6 @@ export default class API {
 	 * Create's 'elementor' api.
 	 */
 	constructor() {
-		window.$e = this;
```
Comment (iNewLegend) [inline modules/web-cli/assets/js/api.js:30]: Why? And what is the alternative?

File: modules/web-cli/assets/js/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import API from './api';
+
+window.$e = new API();
```
Comment (iNewLegend) [inline modules/web-cli/assets/js/index.js:3]: Does the IDE, works well with it?

File: tests/qunit/tests/index.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+	require( './modules/web-cli/assets/js/api/core/commands.spec.js' );
+	require( './modules/web-cli/assets/js/api/core/components.spec.js' );
+	require( './modules/web-cli/assets/js/api/core/data.spec.js' );
+	require( './modules/web-cli/assets/js/api/core/routes.spec.js' );
 
-	require( './core/common/assets/js/api/core/hooks/base.spec.js' );
+	require( './modules/web-cli/assets/js/api/core/hooks/base.spec.js' );
 
-	require( './core/common/assets/js/api/extras/hash-commands.spec' );
+	require( './modules/web-cli/assets/js/api/extras/hash-commands.spec' );
```
Comment (iNewLegend) [inline tests/qunit/tests/index.js:12]: Maybe create index file for all those tests, that holds only require(s).
`tests.spec.js`

**Comment by iNewLegend** [comment]
Please do not forget, move the DOCS also.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/qunit/tests/index.js (hunk 3 above)
Comment (StyleShit) [reply tests/qunit/tests/index.js:12]: Done

File: modules/web-cli/assets/js/api.js [hunk 4]
```diff
@@ -27,8 +27,6 @@ export default class API {
 	 * Create's 'elementor' api.
 	 */
 	constructor() {
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply modules/web-cli/assets/js/api.js:29]: You're right, done.

**Review by matipojo — CHANGES_REQUESTED** [review]
Please resolve the conflicts

**Pull request merged**